fuel-tx/src/transaction/types/chargeable_transaction.rs

```diff
@@ -265,7 +265,11 @@ mod field {
                     .and_then(|predicate| {
                         self.inputs_offset_at(idx).map(|inputs| inputs + predicate)
                     })
-                    .zip(input.predicate_len().map(bytes::padded_len_usize))
+                    .zip(
+                        input
+                            .predicate_len()
+                            .map(|l| bytes::padded_len_usize(l).unwrap_or(usize::MAX)),
```

Comment:
You can use `and_then` instead of `map` here to keep an `Option` and not need the `unwrap_or`. 

The `usize::MAX` feels arbitrary to me.

Reply:
Using `and_then` here would mean combining together (a) input has no `predicate_len` and (b) that the padding overflows. These are fundamentally different things. The idea behind using `usize::MAX` is that the value was already quite close to it, and impossibly large. Any actual usage of the return value will correctly cause `MemoryOverflow` or similar panic. The alternative would be to return `Option<Result<usize, PanicReason>>` , and that feels a bit excessive. Alternatively we could just Rust-panic on such sizes, but I'm not confident that we actually verify the sizes beforehands.